pytensor/tensor/_linalg/solve/rewriting.py

```diff
@@ -49,11 +52,18 @@ def solve_lu_decomposed_system(A_decomp, b, transposed=False, *, core_solve_op:
             b_ndim=b_ndim,
             transposed=transposed,
         )
+    elif assume_a == "pos":
```

Comment:
What's True in `(A_decomp, True)`? Whether it's upper or lower? Don't we need to know?

Reply:
That's fine, maybe add a comment for future devs?

Reply:
Ok I made it respect whatever the core op was set to. No reason not to.

Reply:
The `ChoSolve` op itself doesn't have a lower flag though

Reply:
Yeah exactly. I brought up that it doesn't have a flag because the nodes should still be merged right? Or do the inputs need to be the same as well? I was thinking cho_solve((A, False), b) and cho_solve((A, True), b) would be the same function (with different inputs ofc)

We could change (A, False) to (A.T, True), but then the inputs still aren't the same. I think we have to be respectful of the user's flags, in case only one half of A is being stored.

Reply:
But the user isn't creating chol_factor nor the chol solve in these rewrites so it doesn't matter ever? Unless I'm missing something your first approach was correct.

Reply:
We probably shouldn't even allow chol_factor, lower=True at the graph level, but always do upper and transpose if the user requested.

It's like the solve transposed, we handle the transpositions symbolically to keep less variations floating around.

Reply:
We can just transpose A in that case. The issue is one of merging less options  What happens now if user has a Solve with (A, lower=True), and another with (A.T, lower=False).

Are we merging it here correctly? You were ignoring the transpose info coming from the rewrite. Not that if we never represented one of the forms we couldn't have the two forms in the final graph